cmd/bucket-metadata.go

```diff
@@ -373,3 +376,44 @@ func deleteBucketMetadata(ctx context.Context, obj objectDeleter, bucket string)
 	}
 	return nil
 }
+
+// migrate config for remote targets by encrypting data if currently unencrypted
+func (b *BucketMetadata) migrateTargetConfig(ctx context.Context, objectAPI ObjectLayer) error {
+	var (
+		data []byte
+		err  error
+	)
+	if len(b.BucketTargetsConfigJSON) == 0 {
+		return nil
+	}
+	// Is rotating of creds requested?
+	var reencrypt bool
+	if globalOldCred.IsValid() {
+		data, err = decryptData(b.BucketTargetsConfigJSON, globalOldCred, globalActiveCred)
```

Comment:
This will not work if input is unencrypted.

Reply:
agree this approach is not very ideal - @harshavardhana  was suggesting encrypting only when kms is configured, will need to see how that will work with need to save extra encryption metadata. Changing this PR to wip for now.